docs/production-deployment.rst

```diff
@@ -0,0 +1,488 @@
+that you need to add to customize it. You can read about all the args and ways you can build the image
+in the `<#production-image-build-arguments>`_ chapter below.
+
+Here just a few examples are presented which should give you general understanding what you can customize.
+
+This builds the production image in version 3.7 with additional airflow extras from 1.10.10 Pypi package and
+additional apt dev and runtime dependencies.
+
+.. code-block:: bash
+
+  docker build . \
+    --build-arg PYTHON_BASE_IMAGE="python:3.7-slim-buster" \
+    --build-arg PYTHON_MAJOR_MINOR_VERSION=3.7 \
+    --build-arg AIRFLOW_INSTALL_SOURCES="apache-airflow" \
+    --build-arg AIRFLOW_INSTALL_VERSION="==1.10.12" \
+    --build-arg AIRFLOW_CONSTRAINTS_REFERENCE="constraints-1-10" \
+    --build-arg AIRFLOW_SOURCES_FROM="empty" \
+    --build-arg AIRFLOW_SOURCES_TO="/empty" \
+    --build-arg ADDITIONAL_AIRFLOW_EXTRAS="jdbc"
+    --build-arg ADDITIONAL_PYTHON_DEPS="pandas"
+    --build-arg ADDITIONAL_DEV_DEPS="gcc g++"
+    --build-arg ADDITIONAL_RUNTIME_DEPS="default-jre-headless"
+    --tag my-image
+
+
+the same image can be built using ``breeze`` (it supports auto-completion of the options):
+
+.. code-block:: bash
+
+  ./breeze build-image \
+      --production-image  --python 3.7 --install-airflow-version=1.10.12 \
+      --additional-extras=jdbc --additional-python-deps="pandas" \
+      --additional-dev-deps="gcc g++" --additional-runtime-deps="default-jre-headless"
+
+Customizing & extending the image together
+..........................................
+
+You can combine both - customizing & extending the image. You can build the image first using
+``customize`` method (either with docker command or with ``breeze`` and then you can ``extend``
+the resulting image using ``FROM:`` any dependencies you want.
+
+External sources for dependencies
+---------------------------------
+
+In corporate environments, there is often the need to build your Container images using
+other than default sources of dependencies. The docker file uses standard sources (such as
+Debian apt repositories or PyPI repository. However in corporate environments, the dependencies
+are often only possible to be installed from internal, vetted repositories that are reviewed and
+approved by the internal security teams. In those cases, you might need to use those different
+sources.
+
+This is rather easy if you extend the image - you simply write your extension commands
+using the right sources - either by adding/replacing the sources in apt configuration or
+specifying the source repository in pip install command.
+
+It's a bit more involved in case of customizing the image. We do not have yet (but we are working
+on it) a capability of changing the sources via build args. However, since the builds use
+Dockerfile that is a source file, you can rather easily simply modify the file manually and
+specify different sources to be used by either of the commands.
```

Comment:
```suggestion
This is rather easy if you extend the image - you simply write your extension commands
using the right sources - either by adding/replacing the sources in apt configuration or
specifying the source repository in pip install command.

It's a bit more involved in the case of customizing the image. We do not have yet (but we are working
on it) a capability of changing the sources via build args. However, since the builds use
Dockerfile that is a source file, you can rather easily simply modify the file manually and
specify different sources to be used by either of the commands.
```